### DISCUSSION TITLE
How do I know if my imagemagick supports heic files?

### DISCUSSION THREAD

**Discussion by Bluebuster116** [ImageMagick/ImageMagick#4354] [General]
I've been using imagemagick sporadically over the last three years and honestly since it fills my needs with no issues I haven't bothered to ask much. However lately my mom has been sending me heic files and I can't open them. So I wanted to use imagemagick to convert these files and I can't. As I am using Windows 7. I can't say or know if my version does support this format. Could someone help me?.

**Comment by snibgo** [comment]
What version of IM?

Use the command "magick -list format" and look for a line that starts with "HEIC", like this:

```
      HEIC* HEIC      r--   High Efficiency Image Format (1.12.0)
```
So my installation (the pre-built Windows binary v7.1.0-4) can read, but not write, HEIC files.

If your IM is v6, use "convert" instead of "magick".

**Reply by Bluebuster116** [reply]
@snibgo

Well I did used the following command

`convert -version`

and I got this

C:\Users\Flor>convert -version
Version: ImageMagick 6.9.3-2 Q16 x86 2016-01-23 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2015 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 180040629
Features: Cipher DPC HDRI Modules OpenMP
Delegates (built-in): bzlib cairo freetype jng jp2 jpeg lcms lqr openexr pangoca
iro png ps rsvg tiff webp xml zlib

Thus I believe my current imagemagick version does not support heic images. But since my current windows 7 version is what I use for work, then I have to find a x86 compatible latest imagemagick for that. Does the latest imagemagick support heic images either writing and reading?

**Comment by snibgo** [comment]
> Does the latest imagemagick support heic images either writing and reading?

I expect so. The latest is 7.1.0-10. I use the pre-built binary v7.1.0-4, which supports HEIC for reading. Likewise v6.9.9-50.

**Reply by Bluebuster116** [reply]
@snibgo

So is that a yes?. I mean for writing files. If you don't mind, what command should I use for convert these files?

Is it convert or is it magick?. This part is confusing as it seems there has been some changes between the version which I was using to the one which is the most recent. Can you attend these questions please?. What if i want to reduce the size let's say to a 50 percent, what command would I use?. Can you put one example or a lossless convertion and one which has scaled down to 50 percent of its size?

**Reply by fmw42** [reply]
I believe if you have a current version of libheic, you should be able to read and write. But you would have to check your 

magick -list format

to see if it says r or rw.

For Imagemagick 7, use magick, not convert and not magick convert.  For other tools preface them with magick, such as magick identify and magic mogrify, etc. (But not for convert).

**Comment by Bluebuster116** [comment]
@fmw42

As I mentioned above, I am confused about the code which is now used for converting images from one format to another.

What I've been using is as follows;

Let's say from png to pdf

convert "C:\Users\Flor\Documents\photos_chrome\ethernet.png" "C:\Users\Flor\Documents\photos_chrome\ethernet_720.pdf"

Let's say resizing

convert "C:\Users\Flor\Documents\photos_chrome\ethernet.png"  -resize 720  "C:\Users\Flor\Documents\photos_chrome\ethernet_720.png"

Will those work for the current imagemagick?. It would be nice if you included some example code please.

**Reply by fmw42** [reply]
If on Imagemagick 6, that is correct syntax. If on Imagemagick 7, change convert to magick.

Imagemagick 6:

`convert "C:\Users\Flor\Documents\photos_chrome\ethernet.png" -resize 720 ^ "C:\Users\Flor\Documents\photos_chrome\ethernet_720.png"`

Imagemagick 7

`magick "C:\Users\Flor\Documents\photos_chrome\ethernet.png" -resize 720 ^ "C:\Users\Flor\Documents\photos_chrome\ethernet_720.png"`

The ^ is a new line character to indicate that the command is split into to two line. If all your code for a command is on one line, then remove the ^
